identity/aziot-identityd/src/main.rs

```diff
@@ -60,26 +70,23 @@ async fn run() -> Result<(), Box<dyn std::error::Error>> {
 			if prev_device_info_path.exists() {
 				let prev_hub_device_info = aziot_identityd::settings::HubDeviceInfo::new(&prev_device_info_path)?;
 
-				if let Some(prev_state) = prev_hub_device_info {
-					let mut prev_module_set: std::collections::BTreeSet<aziot_identity_common::ModuleId> = std::collections::BTreeSet::default();
-					if prev_state.eq(&curr_hub_device_info) && prev_settings_path.exists() {
-						let settings = aziot_identityd::settings::Settings::new(&prev_settings_path)?;
-						let (_, p) = convert_to_map(&settings.principal);
-						prev_module_set = p;
+				if let Some(prev_device) = prev_hub_device_info {
+					if prev_device == curr_hub_device_info {
+						let () = server_.init_hub_identities(prev_hub_mset, hub_mset).await?;
```

Comment:
`init_hub_identities` should happen regardless of the `prev_device` and `curr_hub_device_info` match. The prev_module_set was only set if `prev_device` and `curr_hub_device_info` matched.

Reply:
Fixed

Reply:
Note that I've changed the logic so that init_hub_identities will be called if the device is currently provisioned. Before, it was called if the device was provisioned AND the device was previously provisioned.